### PR TITLE
feat[ft-search-product]

### PR DESCRIPTION
**Pull request by ngoma3**
### What does this PR do

-  allow user to search product
-   write its swagger documentation
### How to run it

- git clone from repo
- git checkout ft-search-product
- set .env as shown in .envexample
-  run npm i
-  run npm run dev
-  copy this url in your postman: http://localhost:{port}/api/v1/product
-  you will add query keywords on this GET url `ex`: http://localhost:{port}/api/v1/product?q=apple
-  the api will search for the alike content in the product collection depending on name,description and model
-  it will respond the arrayList with all matched products.

### CODE REVIEW

File: src/controllers/products/productController.js [hunk 1]
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
     message: product.name + ' ' + req.t('productDeleted'),
   });
 });
+export const search = async (req, res) => {
+  console.log("this is where the error should");
+  const { q } = req.query;
+  console.log("this is where the error should occur",q);
```
Comment (Kevin-ishimwe) [inline src/controllers/products/productController.js]: and this one to

File: src/controllers/products/productController.js [hunk 2]
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
     message: product.name + ' ' + req.t('productDeleted'),
   });
 });
+export const search = async (req, res) => {
+  console.log("this is where the error should");
```
Comment (Kevin-ishimwe) [inline src/controllers/products/productController.js]: if possible could we remove these console.log()

**Review by Kevin-ishimwe — APPROVED** [review]

File: src/controllers/products/productController.js [hunk 3]
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
...
+    where: {
+      [Op.or]: [
+        { name: { [Op.iLike]: `%${q}%` } },
+        { description: { [Op.iLike]: `%${q}%` } },,
+        { model: { [Op.iLike]: `%${q}%` } },
+      ],
+    },
+  });
+
+  return res.send({status: req.t('success'), itemsList: items});
```
Comment (FabCode67) [inline src/controllers/products/productController.js]: add status code here?

File: src/controllers/products/productController.js [hunk 4]
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
...
+} catch (err) {
```
Comment (FabCode67) [inline src/controllers/products/productController.js]: remove this catch block  by importing asyncWrapper middleware

**Review by Evaristekanova — DISMISSED** [review]
@ngoma3 try to remove try and catch and use the reusable function to handle that and work on the way you are sending responses cz you are not sending status try to use jsend response

**Review by Christabll — COMMENTED** [review]
I think You should  have proper status message

**Comment by BenoitIT** [comment]
There is a conflict on your PR in reademe

**Review by FabCode67 — APPROVED** [review]
Good job @ngoma3

**Review by BenoitIT — APPROVED** [review]

**Review by Aimeana100 — APPROVED** [review]
Good job @ngoma3

**Review by mansurissa — APPROVED** [review]
LGTM

**Pull request merged**